### DISCUSSION TITLE
Override base branches for previous package rules

### DISCUSSION THREAD

**Discussion by noahnu** [renovatebot/renovate#28538] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I have a preset shared among multiple GitHub repositories with the following `packageRules`:

```json
{
    "groupName": "3rd Party Dev Dependencies (Patch)",
    "matchManagers": ["npm"],
    "matchUpdateTypes": ["patch"],
    "matchCurrentVersion": "!/^0/",
    "matchDepTypes": ["devDependencies"],
},
```

I then have a `renovate.json` file in a consuming repository that extends the base. In this particular repository, I want to force all npm updates to target the `develop` git branch, while all other updates should target the `main` branch. This is my config:

```json
{
    "extends": ["github>MyOrg/renovate-config//presets/base.json5"],
    "baseBranches": ["main", "develop"],
    "packageRules": {
          {
              "matchManagers": ["github-actions"],
              "matchBaseBranches": ["main"],
          },
  
          {
              "matchManagers": ["npm"],
              "matchBaseBranches": ["develop"],
          }
    }
}
```

The problem is that I'm still seeing a PR for "3rd Party Dev Dependencies (Patch)" created against "main" instead of "develop". Is it possible to achieve the behaviour I'm looking for? i.e. all npm changes should target develop and all github-actions changes should target main.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Your packageRules don't do anything. They say "if matching X and Y" and no "then do Z".

By default Renovate will update everything in both base branches. So if you want to change that, you need to add some rules with enabled=false in order to disable some. e.g. want npm only on develop branch? You need to disable them on main
